### DISCUSSION TITLE
webpack5 will compile 3 times

### DISCUSSION THREAD

**Discussion by lanmch** [webpack/webpack#14552] [Q&A]
i just open the cache property, why it will build too many times. And also, when the log tell me that 'webpack build successful', but i can't open the localhost:3001/index.html until a log about webpack cache comes

**Comment by lanmch** [comment]
also, i want to know about the webpack 5 cache, Is it based on babel?

**Comment by alexander-akait** [comment]
Can you clarify?

**Reply by lanmch** [reply]
> Can you clarify?

clarify what?

**Reply by alexander-akait** [reply]
> i just open the cache property

What is `cache` property?

> why it will build too many times

What build? Size, number of modules, configuration? Babel/ts/something else?

> i can't open the localhost:3001/index.html until a log about webpack cache comes

What do you mean "i can't open"? You can't open browser?

> based on babel

No, because you can be without babel, I don't understand how babel related to webpack cache

**Reply by lanmch** [reply]
```json
cache: {
      type: 'filesystem',
      allowCollectingMemory: true,
      cacheDirectory: path.resolve(__dirname, '../.temp_cache'),
    },
```
this is my cache config

**Reply by lanmch** [reply]
What do you mean, because there are babel/ts-loader/other configurations at the same time, it will be compiled three times?

**Reply by lanmch** [reply]
> What do you mean "i can't open"? You can't open browser?

yes ， the output folder has been generated, and the webpack log also tell me that build successful, but i cant open it in browser.
After some seconds, I get a log about cache log, and now it can be opened
![image](https://user-images.githubusercontent.com/37408756/138459935-f85acc0e-ff1d-4493-a31d-d8c392ac1ae6.png)

**Reply by alexander-akait** [reply]
What is `webpack` version?

**Reply by lanmch** [reply]
5.51.1

**Reply by alexander-akait** [reply]
Please update to the latest stable

**Reply by alexander-akait** [reply]
If it will not help can you provide full screenshot of error(s), cache is not working because something wrong with file, so webpack can't cache it

**Reply by lanmch** [reply]
ok I will try to update to the latest stable
